eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -240,4 +233,27 @@ object NodeRelayer {
     )
   }
 
+  /**
+   * This helper method translates relaying errors (returned by the downstream nodes) to a BOLT 4 standard error that we
+   * should return upstream.
+   */
+  private def translateError(failures: Seq[PaymentFailure], outgoingNodeId: PublicKey): Option[FailureMessage] = {
+    def tooManyRouteNotFound(failures: Seq[PaymentFailure]): Boolean = {
+      val routeNotFoundCount = failures.count(f => f == LocalFailure(RouteNotFound))
+      routeNotFoundCount > failures.length / 2
+    }
+
+    failures match {
+      case Nil => None
+      case LocalFailure(MultiPartPaymentLifecycle.BalanceTooLow) :: Nil => Some(TemporaryNodeFailure) // we don't have enough outgoing liquidity at the moment
+      case _ if tooManyRouteNotFound(failures) => Some(TrampolineFeeInsufficient) // if we couldn't find routes, it's likely that the fee/cltv was insufficient
```

Comment:
If I understand correctly, having more than 50% of `RouteNotFound` would mean that we should increase the trampoline fees for this payment to pass. How safe is that 50% estimate in terms of false positives?

Reply:
It's honestly impossible to say. But we need to impose some kind of threshold to get a short-term solution. We'll be able to greatly improve that and provide accurate retry fees/cltv once MPP is directly integrated inside the router, but until then we can only do heuristics.
Note that when that happens we can't know yet if what needs to be raised is fees, cltv or both. So for now Phoenix should slightly raise both (but for CLTV we can keep the upper bound described [here](https://github.com/ACINQ/phoenix/issues/1)).

Reply:
Another thing we could do that would greatly improve the Phoenix-to-non-Phoenix payments is to temporarily disable MPP between Acinq and the recipient for those.
Then there won't be any false positives.
I don't expect many people to do payments that really require MPP in the next couple of months, so that's probably the safe choice for now. I could do that in this PR, it's a simple change. WDYT?